sklearn/utils/tests/test_extmath.py

```diff
@@ -453,6 +455,112 @@ def naive_log_logistic(x):
     assert_array_almost_equal(log_logistic(extreme_x), [-100, 0])
 
 
+def test_incremental_weighted_mean_and_variance_simple():
+    rng = np.random.RandomState(42)
+    mult = 10
+    X = rng.rand(1000, 20)*mult
+    sample_weight = rng.rand(X.shape[0]) * mult
+    mean, var, _ = _incremental_weighted_mean_and_var(X, sample_weight,
+                                                      0, 0, 0)
+
+    mean_exp = np.average(X, weights=sample_weight, axis=0)
+    var_exp = np.average(X**2, weights=sample_weight, axis=0) - mean_exp**2
+    assert_almost_equal(mean, mean_exp)
+    assert_almost_equal(var, var_exp)
+
+
+def test_incremental_weighted_mean_and_variance():
+
+    # Testing of correctness and numerical stability
+    def test(X, sample_weight, mean_exp, var_exp):
+        n = X.shape[0]
+        for chunk_size in [1, n//10 + 1, n//4 + 1, n//2 + 1, n]:
+            last_mean, last_weight_sum, last_var = 0, 0, 0
+            for batch in gen_batches(n, chunk_size):
+                last_mean, last_var, last_weight_sum = \
+                    _incremental_weighted_mean_and_var(X[batch],
+                                                       sample_weight[batch],
+                                                       last_mean,
+                                                       last_var,
+                                                       last_weight_sum)
+            assert_allclose(last_mean, mean_exp)
+            assert_allclose(last_var, var_exp, atol=1e-6)
+
+    high_mean = 1e7
+    low_var = 1e-8
+    high_var = 1e5
+    normal_mean = 0.0
```

Comment:
I think the code below would be clearer if you just had 0 and 1 instead of normal_mean and normal_var; the meaning of `scale=1` and `loc=0` is clear, while the use of "normal" is confusing in this context.